src/main/kotlin/io/github/moulberry/notenoughupdates/miscfeatures/HotmDesires.kt

```diff
@@ -0,0 +1,125 @@
+        val powderRequirement: Int,
+    )
+
+    val desires: MutableMap<String, Desire>?
+        get() = NotEnoughUpdates.INSTANCE.config.profileSpecific?.hotmDesires
+
+    @JvmStatic
+    fun appendDesireForType(powderType: String): String {
+        val desire = desires?.get(powderType) ?: return ""
+        return "§7/§c" + StringUtils.formatNumber(desire.powderRequirement)
+    }
+
+    val tablistPowderLine = " (.*): ([0-9,]+)".toPattern()
+
+    @SubscribeEvent
+    fun onTabListChange(event: TabListChangeEvent) {
+        val desireMap = desires ?: return
+        if (!isEnabled()) {
+            desireMap.clear()
+            return
+        }
+        if (SBInfo.getInstance().getLocation() !in setOf("mining_3", "crystal_hollows", "mineshaft")) {
+            return
+        }
+        for (line in TablistAPI.getWidgetLines(TablistAPI.WidgetNames.POWDER)) {
+            val (powderKind, powderCount) = tablistPowderLine.useMatcher(StringUtils.cleanColour(line)) {
+                val powderKind = group(1)
+                val powderCount = group(2).replace(",", "").toInt()
+                powderKind to powderCount
+            } ?: continue
+            val goal = desireMap[powderKind] ?: continue
+            if (goal.powderRequirement > powderCount) continue
+            desireMap.remove(powderKind)
+            Utils.addClickableChatMessage(
+                "§e[NEU] You have enough $powderKind powder to upgrade ${goal.name}§e!",
+                "/hotm",
+                "§eClick to open your Heart of the Mountain to select the next upgrade."
+            )
+        }
+    }
+
+    fun isEnabled() = NotEnoughUpdates.INSTANCE.config.mining.powderTodo
+
+    @SubscribeEvent
+    fun onClickHotmItemThatYouCannotUpgrade(event: SlotClickEvent) {
+        if (Utils.getOpenChestName() != "Heart of the Mountain" || !isEnabled())
+            return
+        val name = ItemUtils.getDisplayName(event.slot.stack) ?: return
+        val lore = ItemUtils.getLore(event.slot.stack)
+        val missingPowderText = lore.lastOrNull() ?: return
+        val powderKind = youArePoorText.useMatcher(missingPowderText) {
+            group(1)
+        } ?: return
+        val powderCount =
+            lore.firstNotNullOfOrNull { powderRequirementText.useMatcher(it) { group(1).replace(",", "").toInt() } }
+                ?: return
+        val desireMap = desires ?: return
+        if (desireMap[powderKind]?.name != name) {
+            desireMap[powderKind] = Desire(name, powderCount)
+        } else {
+            desireMap.remove(powderKind)
+        }
+    }
+
+
+    @SubscribeEvent
+    fun onAfterGuiDraw(event: ItemTooltipEvent) {
+        if (Utils.getOpenChestName() != "Heart of the Mountain" || !isEnabled())
+            return
+        val name = ItemUtils.getDisplayName(event.itemStack) ?: return
+        if (desires?.values?.any { it.name == name } != true) return
+        event.toolTip.add("§e[NEU] Selected this perk as your next goal.")
```

Comment:
Shouldn't it also show a message when it's not selected, so that the user knows the feature exists

Reply:
Feels a bit intrusive, but can be added